dali/internal/event/images/resource-image-impl.cpp

```diff
@@ -76,18 +77,19 @@ ResourceImagePtr ResourceImage::New( const std::string& url, const ImageAttribut
   if( NinePatchImage::IsNinePatchUrl( url ) )
   {
     image = NinePatchImage::New( url );
+    image->mUrl = url;
   }
   else
   {
     image = new ResourceImage();
     image->Initialize();
 
-    // consider the requested size as natural size, 0 means we don't (yet) know it
-    image->mWidth = attributes.GetWidth();
-    image->mHeight = attributes.GetHeight();
-    image->mRequest = image->mImageFactory.RegisterRequest( url, &attributes );
-    image->mTicket = image->mImageFactory.Load( *image->mRequest.Get() );
-    image->mTicket->AddObserver( *image );
+    image->mParameters.size= ImageDimensions::FromFloatVec2( attributes.GetSize() );
+    image->mParameters.scalingMode = attributes.GetScalingMode();
+    image->mParameters.samplingMode = attributes.GetFilterMode();
+    image->mParameters.orientationCorrection = attributes.GetOrientationCorrection();
+    image->mUrl = url;
+    image->Reload(); // reuse code
```

Comment:
How many steps does a ResourceImage need to go through to be initialised and loaded? We should rethink this API.